### PR TITLE
upgrade dependencies

### PR DESCRIPTION
**Pull request by longfangsong**
Reopen #631.
Merging is blocked.
See #632.

### CODE REVIEW

**Comment by hunterlxt** [comment]
Please make ci passed

**Comment by hunterlxt** [comment]
`s/=0.11.3/0.11` is ok

File: rust-toolchain [hunk 1]
```diff
@@ -0,0 +1 @@
+1.42.0
```
Comment (longfangsong) [inline rust-toolchain:1]: This is caused by a [cargo #8258](https://github.com/rust-lang/cargo/issues/8258)

**Comment by longfangsong** [comment]
@hunterlxt PTAL.

**Comment by longfangsong** [comment]
@hunterlxt @nrc Can we merge this now?

**Comment by hunterlxt** [comment]
This pr can only be merged after tikv upgrade

**Comment by nrc** [comment]
I think we can keep this on a branch for the time being?

**Comment by nrc** [comment]
This is now on the branch client-0.1

**Pull request merged**